File: api/app/models/entertainment.rb
class Entertainment < ApplicationRecord

  validates :name, presence: true
  validates :description, :length => { maximum: 140 }
  validates_inclusion_of :category, :in => %w(cinema sports music books games)
  validates_inclusion_of :score, :in => 0..10

  belongs_to :forum, :class_name => 'Forum', :foreign_key => 'id'

  def calculate_new_score(new_score)
    sum = self[:score] * self[:total_votes]
    sum += new_score
    sum / (self.total_votes + 1)
  end

end

File: app/app/src/main/java/com/ratchitech/trainook/models/Entertainment.java
package com.ratchitech.trainook.models;

public class Entertainment {

    private String description;
    //TODO: enum
    private String category;
    private Double score;

    public Entertainment(String description, String category, Double score) {
        this.description = description;
        this.category = category;
        this.score = score;
    }

}
